oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneDocumentMaker.java

```diff
@@ -331,4 +336,69 @@ protected void indexSimilarityBinaries(Document doc, PropertyDefinition pd, Blob
             }
         }
     }
+    
+    @Override
+    protected boolean indexDynamicBoost(Document doc, PropertyDefinition pd, NodeState nodeState, String propertyName) {
```

Comment:
This method seems to assume a specific structure such as that of a node having a bunch of children with properties DYNAMIC_BOOST_TAG_NAME and DYNAMIC_BOOST_TAG_CONFIDENCE. How is it ensured that this method gets called for only where it ought to be called?

Reply:
> a specific structure such as that of a node having a bunch of children with properties DYNAMIC_BOOST_TAG_NAME and DYNAMIC_BOOST_TAG_CONFIDENCE

Yes, exactly.

> How is it ensured that this method gets called for only where it ought to be called?

This is ensured by the index definition. To enable the feature, add a property to be indexed, e.g.:

    /oak:index/damAssetLucene-...
        + indexRules
            + dam:Asset
                + properties
                    + predictedTags
                        - dynamicBoost = true (Boolean)
                        - propertyIndex = true (Boolean)
                        - isRegexp = true (Boolean)
                        - name = jcr:content/metadata/predictedTags/.* (String)

The "dynamicBoost = true" enables this feature. The method is then called for all dam:Asset nodes that have a property jcr:content/metadata/predictedTags/.* (any property). That includes jcr:content/metadata/predictedTags/jcr:primaryType, meaning the code is called when the node predictedTags is added. Unlike how it was done with the IndexFieldProvider, the code is also called if any of property in that node is modified. I don't think that's needed right now however.



dynamicBoost = true (Boolean)
propertyIndex = true
name = 
isRegexp = true (Boolean)

Reply:
Ok. The `dynamicBoost` and `propertyIndex` look like new additions. Below is from the current index definition of `/oak:index/damAssetLucene/indexRules/dam:Asset/properties/predictedTags` 
![Screenshot 2020-04-20 at 6 10 24 PM](https://user-images.githubusercontent.com/1191451/79752770-73a4b900-8332-11ea-9cbb-4bb36554ca96.png)